main.js

```diff
@@ -0,0 +1,910 @@
+                                resolve(res)
+                            }).catch(error => {
+                                console.log(error)
+                                // most likely the infohash of this torrent does not match what we have currently, stop this torrent
+                                self.webtorrent.remove(checkTorrent.infoHash, {destroyStore: false})
+                                resolve(null)
+                            })
+                        })
+                    ])
+                    if(checkProperty){
+                        delete checkProperty.infoHash
+                        for(const prop in checkProperty){
+                            checkTorrent[prop] = checkProperty[prop]
+                        }
+                        checkTorrent.folder = folderPath
+                        checkTorrent.side = true
+                        console.log(checkInternal[i] + ' is good')
+                    }
+                }
+            } else if(checkTitle.test(checkInternal[i])){
+                let checkTorrent = await Promise.any([
+                    new Promise((resolve, reject) => {
+                        setTimeout(() => {resolve(null)}, self._status.timeout)
+                    }),
+                    new Promise((resolve, reject) => {
+                        self.webtorrent.seed(folderPath, {destroyStoreOnDestroy: true}, torrent => {
+                            resolve(torrent)
+                        })
+                    })
+                ])
+                if(checkTorrent){
+                    checkTorrent.folder = folderPath
+                    checkTorrent.title = checkInternal[i]
+                    checkTorrent.side = true
+                    console.log(checkInternal[i] + ' is good')
+                }
+            } else {
+                await fs.remove(folderPath)
+            }
+        }
+    }
+
+    // if share option is true, then start seeding(webtorrent.add seeds after torrent is downloaded) on start up
+    if(self._status.share){
+        let checkExternal = await fs.readdir(self._external, {withFileTypes: false})
+        for(let i = 0;i < checkExternal.length;i++){
+            let folderPath = self._external + path.sep + checkExternal[i]
+            if(checkAddress.test(checkExternal[i])){
+                let checkProperty = await Promise.any([
+                    new Promise((resolve, reject) => {setTimeout(() => {resolve(null)}, self._status.timeout)}),
+                    self.resolve(checkExternal[i])
+                ])
+                if(checkProperty){
+                    if(self._status.current){
+                        if(!await fs.pathExists(folderPath + path.sep + checkProperty.infoHash)){
+                            try {
+                                await fs.emptyDir(folderPath)
+                            } catch (error) {
+                                console.log(error)
```

Comment:
followed the fs-extra docs. this is how they have it when `await` is used. i do not see this ever failing. it creates the directory if it does not exist.

https://github.com/jprichardson/node-fs-extra/blob/HEAD/docs/emptyDir.md

Reply:
Personally I try to let errors bubble up whenever possible, but if you're set on this, it'd be good to use `console.error` instead of `console.log` since STDERR is more appropriate for error messages.